### DISCUSSION TITLE
VRPTW and fixed service times

### DISCUSSION THREAD

**Discussion by juntek** [google/or-tools#2365] [Routing (and legacy CP) questions] [answered]
What is the proper way to set fixed service times for specified locations? 
Once I add them my VRPTW stops adding driving times between locations to the solution.
As a result start time of the next location is increased only by service time at previous location, but driving time between consecutive locations is not added. Here is the code I am using:

```cpp
ort::Solver* solver = routingModel.solver();
std::string ss("Break on vehicle 0");

// vehicle break 45 minutes between hours 1 and 2
ort::IntervalVar* break_interval = solver->MakeFixedDurationIntervalVar(3600, 7200, 2700, false, ss);

std::vector<int64> service_times = { 0, 300, 600, 900 }; // depot 0 + three locations: 5, 10, 15 minutes
timeDimension->SetBreakIntervalsOfVehicle({break_interval}, 0, service_times);
```

**Comment by Mizux** [comment, marked as answer]
1. Your transit(fromA, toB) callback must return `service time(A) + travel time(A, B)`.

2. Since your transit callback return the sum, when adding a break solver must need to know what is the service time part thus the SetBreakIntervals() method needs it.

3. So please show us your time dimension transit callback, which should be close to something like this:
```cpp
int64 time_callback(NodeIndex from_index, NodeIndex to_index) {
  from = manager.IndexToNode(from_index);
  to = manager.IndexToNode(to_index);
  return time_matrix[from][to] + service_times[from];
}
```

**Reply by juntek** [reply]
Aha, now I understand. That's correct, my transit callback did not include service times part because I thought the SetBreakIntervalsOfVehicle() function serves the purpose. After modification it works very well now. Thank you!
